### ISSUE TITLE
20.4 Segmentation fault during query with illegal LowCardinality filter type

### ISSUE THREAD

**Issue by millin** [ClickHouse/ClickHouse#11977]
**Describe the bug**
Segmentation fault occurs when WHERE filter contains illegal LowCardinality type.

**How to reproduce**

- ClickHouse server version 20.4.6.53

- Run the query with illegal filter:

> `SELECT * FROM table WHERE lowcardinality_string_column`

**Expected behavior**
> ` DB::Exception: Illegal type LowCardinality(String) of column for filter. Must be UInt8 or Nullable(UInt8) or Const variants of them..`

**Error message and/or stacktrace**
```
2020.06.26 06:52:26.828440 [ 1177 ] {} <Fatal> BaseDaemon: ########################################                                                                                                                                                          [20/48161]
2020.06.26 06:52:26.828534 [ 1177 ] {} <Fatal> BaseDaemon: (version 20.4.6.53 (official build)) (from thread 707) (query_id: 2b73ed59-7e17-4435-b513-a6a1009f8a8f) Received signal Segmentation fault (11).                                                            
2020.06.26 06:52:26.828569 [ 1177 ] {} <Fatal> BaseDaemon: Address: 0x14 Access: read. Address not mapped to object.                                                                                                                                                   
2020.06.26 06:52:26.828597 [ 1177 ] {} <Fatal> BaseDaemon: Stack trace: 0xcee82f0 0xcee8ad5 0xd5afbbf 0xcf5081d 0xd9bffff 0xd9c4b40 0xd9c5b74 0xd9eae0c 0xd9237d6 0xd92cadc 0xd92f433 0xd74048d 0xd64557c 0xd273c16 0xd276868 0xd23942d 0xd6746a7 0xd674af7 0xd675cbd 0
xd67c9f8 0xd665e75 0xd64557c 0xd273c16 0xd276868 0xd239673 0xd404fc0 0xd4070ec 0xd5445c9 0xd548000                                                                                                                                                                     
2020.06.26 06:52:26.829727 [ 1177 ] {} <Fatal> BaseDaemon: 3. DB::ReverseIndex<unsigned long, DB::ColumnString>::insert(StringRef const&) @ 0xcee82f0 in /usr/bin/clickhouse                                                                                           
2020.06.26 06:52:26.829799 [ 1177 ] {} <Fatal> BaseDaemon: 4. DB::ColumnUnique<DB::ColumnString>::uniqueInsertData(char const*, unsigned long) @ 0xcee8ad5 in /usr/bin/clickhouse                                                                                      
2020.06.26 06:52:26.830809 [ 1177 ] {} <Fatal> BaseDaemon: 5. DB::ColumnLowCardinality::insert(DB::Field const&) @ 0xd5afbbf in /usr/bin/clickhouse                                                                                                                    
2020.06.26 06:52:26.830916 [ 1177 ] {} <Fatal> BaseDaemon: 6. DB::IDataType::createColumnConst(unsigned long, DB::Field const&) const @ 0xcf5081d in /usr/bin/clickhouse                                                                                               
2020.06.26 06:52:26.831798 [ 1177 ] {} <Fatal> BaseDaemon: 7. DB::MergeTreeBaseSelectProcessor::executePrewhereActions(DB::Block&, std::__1::shared_ptr<DB::PrewhereInfo> const&) @ 0xd9bffff in /usr/bin/clickhouse                                                   
2020.06.26 06:52:26.831857 [ 1177 ] {} <Fatal> BaseDaemon: 8. DB::MergeTreeBaseSelectProcessor::getHeader(DB::Block, std::__1::shared_ptr<DB::PrewhereInfo> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char
> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0xd9c4b40 in /usr/bin/clickhouse                                                                                                          
2020.06.26 06:52:26.831896 [ 1177 ] {} <Fatal> BaseDaemon: 9. DB::MergeTreeBaseSelectProcessor::MergeTreeBaseSelectProcessor(DB::Block, DB::MergeTreeData const&, std::__1::shared_ptr<DB::PrewhereInfo> const&, unsigned long, unsigned long, unsigned long, DB::Merge
TreeReaderSettings const&, bool, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0xd9c5
b74 in /usr/bin/clickhouse                                                                                                                                                                                                                                             
2020.06.26 06:52:26.832572 [ 1177 ] {} <Fatal> BaseDaemon: 10. DB::MergeTreeThreadSelectBlockInputProcessor::MergeTreeThreadSelectBlockInputProcessor(unsigned long, std::__1::shared_ptr<DB::MergeTreeReadPool> const&, unsigned long, unsigned long, unsigned long, u
nsigned long, DB::MergeTreeData const&, bool, std::__1::shared_ptr<DB::PrewhereInfo> const&, DB::MergeTreeReaderSettings const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::
basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0xd9eae0c in /usr/bin/clickhouse                                                                                                                                             
2020.06.26 06:52:26.833023 [ 1177 ] {} <Fatal> BaseDaemon: 11. DB::MergeTreeDataSelectExecutor::spreadMarkRangesAmongStreams(std::__1::vector<DB::RangesInDataPart, std::__1::allocator<DB::RangesInDataPart> >&&, unsigned long, std::__1::vector<std::__1::basic_stri
ng<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, unsigned long, bool, DB::SelectQueryInfo const&, std::__1::vector<std::__1::ba
sic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::Settings const&, DB::MergeTreeReaderSettings const&) const @ 0xd9237d6 in /usr/bin/clickhouse
2020.06.26 06:52:26.833108 [ 1177 ] {} <Fatal> BaseDaemon: 12. DB::MergeTreeDataSelectExecutor::readFromParts(std::__1::vector<std::__1::shared_ptr<DB::IMergeTreeDataPart const>, std::__1::allocator<std::__1::shared_ptr<DB::IMergeTreeDataPart const> > >, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, unsigned long, unsigned int, std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, long, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, long> > > const*) const @ 0xd92cadc in /usr/bin/clickhouse
2020.06.26 06:52:26.833912 [ 1177 ] {} <Fatal> BaseDaemon: 13. DB::MergeTreeDataSelectExecutor::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, unsigned long, unsigned int, std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, long, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, long> > > const*) const @ 0xd92f433 in /usr/bin/clickhouse
2020.06.26 06:52:26.833998 [ 1177 ] {} <Fatal> BaseDaemon: 14. DB::StorageReplicatedMergeTree::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0xd74048d in /usr/bin/clickhouse
2020.06.26 06:52:26.834048 [ 1177 ] {} <Fatal> BaseDaemon: 15. DB::IStorage::readStreams(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0xd64557c in /usr/bin/clickhouse
2020.06.26 06:52:26.834096 [ 1177 ] {} <Fatal> BaseDaemon: 16. void DB::InterpreterSelectQuery::executeFetchColumns<DB::InterpreterSelectQuery::Pipeline>(DB::QueryProcessingStage::Enum, DB::InterpreterSelectQuery::Pipeline&, std::__1::shared_ptr<DB::PrewhereInfo> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::QueryPipeline&) @ 0xd273c16 in /usr/bin/clickhouse
2020.06.26 06:52:26.834137 [ 1177 ] {} <Fatal> BaseDaemon: 17. void DB::InterpreterSelectQuery::executeImpl<DB::InterpreterSelectQuery::Pipeline>(DB::InterpreterSelectQuery::Pipeline&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, DB::QueryPipeline&) @ 0xd276868 in /usr/bin/clickhouse
2020.06.26 06:52:26.834159 [ 1177 ] {} <Fatal> BaseDaemon: 18. DB::InterpreterSelectQuery::execute() @ 0xd23942d in /usr/bin/clickhouse
2020.06.26 06:52:26.834180 [ 1177 ] {} <Fatal> BaseDaemon: 19. ? @ 0xd6746a7 in /usr/bin/clickhouse
2020.06.26 06:52:26.834204 [ 1177 ] {} <Fatal> BaseDaemon: 20. ? @ 0xd674af7 in /usr/bin/clickhouse
2020.06.26 06:52:26.834245 [ 1177 ] {} <Fatal> BaseDaemon: 21. DB::ClusterProxy::SelectStreamFactory::createForShard(DB::Cluster::ShardInfo const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::Throttler> const&, DB::SelectQueryInfo const&, std::__1::vector<DB::Pipe, std::__1::allocator<DB::Pipe> >&) @ 0xd675cbd in /usr/bin/clickhouse
2020.06.26 06:52:26.834275 [ 1177 ] {} <Fatal> BaseDaemon: 22. DB::ClusterProxy::executeQuery(DB::ClusterProxy::IStreamFactory&, std::__1::shared_ptr<DB::Cluster> const&, std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::Settings const&, DB::SelectQueryInfo const&) @ 0xd67c9f8 in /usr/bin/clickhouse
2020.06.26 06:52:26.834309 [ 1177 ] {} <Fatal> BaseDaemon: 23. DB::StorageDistributed::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0xd665e75 in /usr/bin/clickhouse
2020.06.26 06:52:26.834333 [ 1177 ] {} <Fatal> BaseDaemon: 24. DB::IStorage::readStreams(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0xd64557c in /usr/bin/clickhouse
2020.06.26 06:52:26.834357 [ 1177 ] {} <Fatal> BaseDaemon: 25. void DB::InterpreterSelectQuery::executeFetchColumns<DB::InterpreterSelectQuery::Pipeline>(DB::QueryProcessingStage::Enum, DB::InterpreterSelectQuery::Pipeline&, std::__1::shared_ptr<DB::PrewhereInfo> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, DB::QueryPipeline&) @ 0xd273c16 in /usr/bin/clickhouse
2020.06.26 06:52:26.834380 [ 1177 ] {} <Fatal> BaseDaemon: 26. void DB::InterpreterSelectQuery::executeImpl<DB::InterpreterSelectQuery::Pipeline>(DB::InterpreterSelectQuery::Pipeline&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pip
e>, DB::QueryPipeline&) @ 0xd276868 in /usr/bin/clickhouse
2020.06.26 06:52:26.834401 [ 1177 ] {} <Fatal> BaseDaemon: 27. DB::InterpreterSelectQuery::executeWithMultipleStreams(DB::QueryPipeline&) @ 0xd239673 in /usr/bin/clickhouse
2020.06.26 06:52:26.834433 [ 1177 ] {} <Fatal> BaseDaemon: 28. DB::InterpreterSelectWithUnionQuery::executeWithMultipleStreams(DB::QueryPipeline&) @ 0xd404fc0 in /usr/bin/clickhouse
2020.06.26 06:52:26.834464 [ 1177 ] {} <Fatal> BaseDaemon: 29. DB::InterpreterSelectWithUnionQuery::execute() @ 0xd4070ec in /usr/bin/clickhouse
2020.06.26 06:52:26.834490 [ 1177 ] {} <Fatal> BaseDaemon: 30. ? @ 0xd5445c9 in /usr/bin/clickhouse
2020.06.26 06:52:26.834515 [ 1177 ] {} <Fatal> BaseDaemon: 31. DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, DB::Context&, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>) @ 0xd548000 in /usr/bin/clickhouse
```

**Comment by alexey-milovidov** [comment]
@KochetovNicolai Looks like already fixed issue?

**Comment by KochetovNicolai** [comment]
It will be fixed in #12060
